test/jb_projects/challenge_handle.test.js

```diff
@@ -0,0 +1,164 @@
+    const { projectOwner, deployer, jbProjectsStore } = await setup();
+
+    await jbProjectsStore
+      .connect(deployer)
+      .createFor(
+        /*owner=*/ projectOwner.address,
+        /*handle=*/ ethers.utils.formatBytes32String(PROJECT_HANDLE),
+        /*metadataCid=*/ METADATA_CID,
+      );
+
+    await jbProjectsStore
+      .connect(projectOwner)
+      .transferHandleOf(
+        /*projectId=*/ PROJECT_ID,
+        /*address=*/ deployer.address,
+        /*handle=*/ ethers.utils.formatBytes32String(PROJECT_HANDLE_NOT_TAKEN),
+      );
+
+    await jbProjectsStore
+      .connect(deployer)
+      .challengeHandle(/*handle=*/ ethers.utils.formatBytes32String(PROJECT_HANDLE_NOT_TAKEN));
+
+    await expect(
+      jbProjectsStore
+        .connect(deployer)
+        .challengeHandle(/*handle=*/ ethers.utils.formatBytes32String(PROJECT_HANDLE_NOT_TAKEN)),
+    ).to.be.revertedWith('0x0e: CHALLENGE_OPEN');
+  });
+
+  // TODO:  Fix this in the protocol.Currently the Protocol allows the
+  //        challenge to happen from the owner of the project that has
+  //        the handle challenged.
+  //
+  // it('Doesn\'t challenge if it is the owner of the project', async function () {
+  //  const { projectOwner, deployer, jbProjectsStore } = await setup();
+
+  //   await jbProjectsStore
+  //     .connect(deployer)
+  //     .createFor(
+  //       /*owner=*/ projectOwner.address,
+  //       /*handle=*/ ethers.utils.formatBytes32String(PROJECT_HANDLE),
+  //       /*metadataCid=*/ METADATA_CID,
+  //     )
+
+  //   await jbProjectsStore
+  //     .connect(projectOwner)
+  //     .transferHandleOf(
+  //         /*projectId=*/ PROJECT_ID,
+  //         /*address=*/ deployer.address,
+  //         /*handle=*/ ethers.utils.formatBytes32String(PROJECT_HANDLE_NOT_TAKEN)
+  //     )
+
+  //   await expect(
+  //     jbProjectsStore
+  //       .connect(projectOwner)
+  //       .challengeHandle(
+  //         /*handle=*/ ethers.utils.formatBytes32String(PROJECT_HANDLE_NOT_TAKEN)
+  //       ),
+  //   ).to.be.revertedWith('0x0e: CHALLENGE_OPEN');
+  // });
+
+  it('Should challenge handle successfully', async function () {
+    const { projectOwner, deployer, jbProjectsStore } = await setup();
+
+    await jbProjectsStore
+      .connect(deployer)
+      .createFor(
+        /*owner=*/ projectOwner.address,
+        /*handle=*/ ethers.utils.formatBytes32String(PROJECT_HANDLE),
+        /*metadataCid=*/ '',
+      );
+
+    await jbProjectsStore
+      .connect(deployer)
+      .createFor(
+        /*owner=*/ deployer.address,
+        /*handle=*/ ethers.utils.formatBytes32String(PROJECT_HANDLE_NOT_TAKEN_2),
+        /*metadataCid=*/ '',
+      );
+
+    await jbProjectsStore
+      .connect(projectOwner)
+      .transferHandleOf(
+        /*projectId=*/ PROJECT_ID,
+        /*address=*/ deployer.address,
+        /*handle=*/ ethers.utils.formatBytes32String(PROJECT_HANDLE_NOT_TAKEN),
+      );
```

Comment:
Same re: test setup. Also, generally the happy path comes first in the tests we've written so far.